evalml/tests/component_tests/test_target_imputer.py

```diff
@@ -153,7 +153,7 @@ def test_target_imputer_fit_transform_all_nan_empty(y):
 def test_target_imputer_numpy_input():
     y = np.array([np.nan, 0, 2])
     imputer = TargetImputer(impute_strategy="mean")
-    y_expected = pd.Series([1, 0, 2], dtype="int64")
+    y_expected = pd.Series([1, 0, 2], dtype="float64")
```

Comment:
This change is interesting to me, do you know why it happened?

Reply:
It's related to the `test_target_imputer_doesnt_truncate_imputed_ints` test I added that you ask about below. 

The target imputer was previously turning all `IntegerNullable` columns to `Integer` after imputing, which would truncate floating point values for the mean and median impute strategies. I confirmed that with the test and then switched it to use the Double logical type in those cases, which is why this now becomes `float64`. It was just coincidence that this example imputing `[nan, 0, 2]` has a whole number mean of 1, but we do want this to be a float (this behavior matches what the simple imputer does).